ntia_conformance_checker/cli_tools/checker.py

```diff
@@ -2,22 +2,43 @@
 
 # pylint: disable=import-error
 
+import json
+
 import click
 from check_anything import check_minimum_elements
+from output import structure_messages
 
 
-@click.command()
+@click.command(context_settings={"show_default": True})
 @click.option("--file", prompt="File name", help="The file to be parsed")
-def main(file):
+@click.option(
+    "--output",
+    default="print",
+    type=click.Choice(["print", "json"]),
+    help="Output format",
+)
+@click.option(
+    "--output_path",
+    help="Filepath for output of JSON.",
+)
+def main(file, output, output_path):
     """
     COMMAND-LINE TOOL that checks for NTIA's minimum elements within a
     file of RDF, XML, JSON, YAML or XML format.
 
-    To use : run `python3 checker.py` using terminal or run
-    `python3 checker.py --file <file name>`
-
+    For help: run `python3 checker.py --help`
     """
-    print(check_minimum_elements(file).messages)
+    if output == "print":
+        print(check_minimum_elements(file).messages)
+    if output == "json":
+        msgs = check_minimum_elements(file).messages
+        result_dict = structure_messages(file, msgs)
+        print(json.dumps(result_dict, indent=2))
```

Comment:
Did we want to print to the screen if an output path is specified?

I would expect if a file output is specified, it would not also print to the screen.

We could also generalize the output path to also be used for the text results.  I'm not sure if it would be useful or not, but it would add some flexibility to the usage.

Reply:
On whether to print to the screen if a file output is specified: I also would expect that that IF an output file is specified, the program would NOT print to the screen. Let me revise that.

And on whether to generalize the output path: that sounds like a good idea too. Let me also revise that.

Reply:
commit [407d586](https://github.com/spdx/ntia-conformance-checker/pull/23/commits/407d586cf9b44adf292386433a8515cd97b19dfd) fixed the printing logic.

commit [e71fbe1](https://github.com/spdx/ntia-conformance-checker/pull/23/commits/e71fbe18998994643a7a05dfcac9bc63fb1076ba) generalized the output file logic.

TY @goneall